dags/process_elife_zip_dag.py

```diff
@@ -138,30 +142,32 @@ def update_tiff_references_to_jpeg_in_article(**context):
     article_bytes = BytesIO(response['Body'].read())
     article_xml = etree.parse(article_bytes)
 
-    # save a copy of the original xml document before making modifications
-    key = s3_key.replace('.xml', '-original.xml')
-    s3.put_object(Bucket=DESTINATION_BUCKET, Key=key, Body=article_bytes.getvalue())
-    logger.info('%s uploaded to %s/%s', key, DESTINATION_BUCKET, folder_name)
-
-    for element in article_xml.xpath(
-            '//*[@mimetype="image" and @mime-subtype="tiff"]'):
-        href = '{http://www.w3.org/1999/xlink}href'
-        element.attrib[href] = re.sub(r'\.\w+$', '.jpg', element.attrib[href])
-        element.attrib['mime-subtype'] = 'jpeg'
-
-    # upload modified document
-    s3.put_object(
-        Bucket=DESTINATION_BUCKET,
-        Key=s3_key,
-        Body=etree.tostring(article_xml, xml_declaration=True, encoding='UTF-8')
-    )
-    logger.info('%s uploaded to %s/%s', s3_key, DESTINATION_BUCKET, folder_name)
+    matches = article_xml.xpath('//*[@mimetype="image" and @mime-subtype="tiff"]')
+    if matches:
+        for element in matches:
+            href = '{http://www.w3.org/1999/xlink}href'
+            element.attrib[href] = re.sub(r'\.\w+$', '.jpg', element.attrib[href])
+            element.attrib['mime-subtype'] = 'jpeg'
+
+        # upload modified document
+        s3_key = s3_key.replace('.xml', '-tiff_to_jpeg.xml')
+        s3.put_object(
+            Bucket=DESTINATION_BUCKET,
+            Key=s3_key,
+            Body=etree.tostring(article_xml, xml_declaration=True, encoding='UTF-8')
+        )
+        logger.info('%s uploaded to %s/%s', s3_key, DESTINATION_BUCKET, folder_name)
+
+    return s3_key
 
 
 def wrap_article_in_libero_xml_and_send_to_service(**context):
-    zip_file_name = get_file_name_passed_to_dag_run_conf_file(**context)
-    article_name = get_expected_elife_article_name(zip_file_name)
-    s3_key = zip_file_name.replace('.zip', '/') + article_name
+    s3_key = get_return_value_from_previous_task(**context)
+    logger.info('ARTICLE S3 KEY PASSED FROM PREVIOUS TASK= %s', s3_key)
```

Comment:
why the uppercase? Just standing out?

Reply:
Correct. There are a lot of logs, especially if you enable debug mode.